<ui version="4.0" >
 <class>VCButtonProperties</class>
 <widget class="QDialog" name="VCButtonProperties" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>397</width>
    <height>516</height>
   </rect>
  </property>
  <property name="windowTitle" >
   <string>Button properties</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout" >
   <item>
    <widget class="QGroupBox" name="groupBox" >
     <property name="title" >
      <string>General</string>
     </property>
     <layout class="QGridLayout" >
      <item row="0" column="0" >
       <widget class="QLabel" name="m_nameLabel" >
        <property name="text" >
         <string>Button label</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1" colspan="3" >
       <widget class="QLineEdit" name="m_nameEdit" >
        <property name="toolTip" >
         <string>Text to display on the button</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0" >
       <widget class="QLabel" name="m_functionLabel" >
        <property name="text" >
         <string>Function</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1" >
       <widget class="QLineEdit" name="m_functionEdit" >
        <property name="toolTip" >
         <string>The function that this button controls</string>
        </property>
        <property name="readOnly" >
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="2" >
       <widget class="QToolButton" name="m_attachFunction" >
        <property name="toolTip" >
         <string>Attach a function to this button</string>
        </property>
        <property name="text" >
         <string>...</string>
        </property>
        <property name="icon" >
         <iconset resource="main.qrc" >
          <normaloff>:/attach.png</normaloff>:/attach.png</iconset>
        </property>
        <property name="iconSize" >
         <size>
          <width>26</width>
          <height>26</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="1" column="3" >
       <widget class="QToolButton" name="m_detachFunction" >
        <property name="toolTip" >
         <string>Detach the button's function attachment</string>
        </property>
        <property name="text" >
         <string>...</string>
        </property>
        <property name="icon" >
         <iconset resource="main.qrc" >
          <normaloff>:/detach.png</normaloff>:/detach.png</iconset>
        </property>
        <property name="iconSize" >
         <size>
          <width>26</width>
          <height>26</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2" >
     <property name="title" >
      <string>Keyboard hotkey</string>
     </property>
     <layout class="QHBoxLayout" >
      <item>
       <widget class="QLabel" name="m_keyLabel" >
        <property name="text" >
         <string>Key combination</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="m_keyEdit" >
        <property name="toolTip" >
         <string>Keyboard combination that toggles this button</string>
        </property>
        <property name="readOnly" >
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="m_attachKey" >
        <property name="toolTip" >
         <string>Set a key combination for this button</string>
        </property>
        <property name="text" >
         <string>...</string>
        </property>
        <property name="icon" >
         <iconset resource="main.qrc" >
          <normaloff>:/key_bindings.png</normaloff>:/key_bindings.png</iconset>
        </property>
        <property name="iconSize" >
         <size>
          <width>26</width>
          <height>26</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="m_detachKey" >
        <property name="toolTip" >
         <string>Remove the button's keyboard shortcut key</string>
        </property>
        <property name="text" >
         <string>...</string>
        </property>
        <property name="icon" >
         <iconset resource="main.qrc" >
          <normaloff>:/fileclose.png</normaloff>:/fileclose.png</iconset>
        </property>
        <property name="iconSize" >
         <size>
          <width>26</width>
          <height>26</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_3" >
     <property name="title" >
      <string>External Input</string>
     </property>
     <layout class="QGridLayout" name="gridLayout" >
      <item row="0" column="0" >
       <widget class="QLabel" name="label_4" >
        <property name="text" >
         <string>Input Universe</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1" colspan="2" >
       <widget class="QLineEdit" name="m_inputUniverseEdit" >
        <property name="toolTip" >
         <string>The input universe that sends data to this widget</string>
        </property>
        <property name="readOnly" >
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0" >
       <widget class="QLabel" name="label_5" >
        <property name="text" >
         <string>Input Channel</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1" colspan="2" >
       <widget class="QLineEdit" name="m_inputChannelEdit" >
        <property name="toolTip" >
         <string>The particular input channel within the input universe that sends data to this widget</string>
        </property>
        <property name="readOnly" >
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="0" colspan="2" >
       <spacer name="horizontalSpacer" >
        <property name="orientation" >
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0" >
         <size>
          <width>268</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="2" column="2" >
       <widget class="QPushButton" name="m_chooseInputButton" >
        <property name="sizePolicy" >
         <sizepolicy vsizetype="Fixed" hsizetype="Maximum" >
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip" >
         <string>Choose the external input universe &amp; channel that this widget should listen to</string>
        </property>
        <property name="text" >
         <string>Choose...</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_4" >
     <property name="title" >
      <string>On button press...</string>
     </property>
     <layout class="QVBoxLayout" >
      <item>
       <widget class="QRadioButton" name="m_toggle" >
        <property name="toolTip" >
         <string>Toggle the assigned function on/off with this button</string>
        </property>
        <property name="text" >
         <string>Toggle function on/off</string>
        </property>
        <property name="checked" >
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="m_flash" >
        <property name="toolTip" >
         <string>Flash the assigned function with this button</string>
        </property>
        <property name="text" >
         <string>Flash function</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_5" >
     <property name="title" >
      <string>Panic</string>
     </property>
     <layout class="QVBoxLayout" >
      <item>
       <widget class="QCheckBox" name="m_stopFunctionsCheck" >
        <property name="toolTip" >
         <string>Stop all running functions when this button is clicked</string>
        </property>
        <property name="text" >
         <string>Stop all running functions</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="m_buttonBox" >
     <property name="orientation" >
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons" >
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="main.qrc" />
 </resources>
 <connections>
  <connection>
   <sender>m_buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>VCButtonProperties</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel" >
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>VCButtonProperties</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel" >
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
